### PR TITLE
Send to taproot addresses

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: backend/coins/btc/util/util_test.go [hunk 1]
```diff
@@ -60,6 +61,16 @@ func TestPkScriptFromAddress(t *testing.T) {
...
 		pkScript)
+
+	// Taproot: test vector #1 from https://github.com/bitcoin/bips/blob/d7cc20992724ea484087c07d2ed53fb3bc3a108b/bip-0086.mediawiki#test-vectors
+	pubkey := []byte("\xa6\x08\x69\xf0\xdb\xcf\x1d\xc6\x59\xc9\xce\xcb\xaf\x80\x50\x13\x5e\xa9\xe8\xcd\xc4\x87\x05\x3f\x1d\xc6\x88\x09\x49\xdc\x68\x4c")
+	address, err = btcutil.NewAddressTaproot(pubkey, net)
+	require.NoError(t, err)
+	pkScript, err = PkScriptFromAddress(address)
+	require.NoError(t, err)
+	require.Equal(t,
+		[]byte("\x51\x20\xa6\x08\x69\xf0\xdb\xcf\x1d\xc6\x59\xc9\xce\xcb\xaf\x80\x50\x13\x5e\xa9\xe8\xcd\xc4\x87\x05\x3f\x1d\xc6\x88\x09\x49\xdc\x68\x4c"),
```
Comment (x1ddos) [inline backend/coins/btc/util/util_test.go:72]: would've been so much easier to compare with 
image

if it were typed as a regular string and then `hex.DecodeString`-decoded for require.Equal

**Review by x1ddos — COMMENTED** [review]
to spare you a couple clicks, a linter doesn't like some deadcode:

```
+ golangci-lint run
backend/devices/bitbox02/protobuf.go:39:5: `btcMsgOutputTypeMap` is unused (deadcode)
var btcMsgOutputTypeMap = map[txscript.ScriptClass]messages.BTCOutputType{
    ^
```

File: backend/coins/btc/addresses/address.go [hunk 2]
```diff
@@ -126,7 +127,7 @@ func (address *AccountAddress) isUsed() bool {
 
 // PubkeyScript returns the pubkey script of this address. Use this in a tx output to receive funds.
 func (address *AccountAddress) PubkeyScript() []byte {
-	script, err := txscript.PayToAddrScript(address.Address)
+	script, err := util.PkScriptFromAddress(address.Address)
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/coins/btc/addresses/address.go:130]: too bad there is already `github.com/btcsuite/btcutil` in the imports :O

File: backend/coins/btc/util/util_test.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	net := &chaincfg.MainNetParams
+
+	var address btcutil.Address
+
+	address, err := btcutil.NewAddressPubKeyHash(hash, net)
+	require.NoError(t, err)
+	pkScript, err := PkScriptFromAddress(address)
+	require.NoError(t, err)
+	require.Equal(t,
+		[]byte("\x76\xa9\x14\x92\x95\x3b\x69\x91\x29\x70\x02\xfa\xa6\x2a\x1d\xd2\x43\x13\xff\x62\x1e\x10\xab\x88\xac"),
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/coins/btc/util/util_test.go:37]: No BIPS or test vectors I am aware of, though they are produced by `txscript.PayToAddrScript()` from btcd, which I assume are unit tested independently. I added the test to check that this refactor into a util function works the same as before.

File: backend/coins/btc/util/util.go [hunk 4]
```diff
@@ -55,6 +65,14 @@ func PkScriptFromAddress(address btcutil.Address) ([]byte, error) {
 
 // AddressFromPkScript decodes a pkScript into an Address instance.
 func AddressFromPkScript(pkScript []byte, net *chaincfg.Params) (btcutil.Address, error) {
+	// Parse taproot scripts. Drop this check once `ExtractPkScriptAddrs` does this upstream.
+	// https://github.com/bitcoin/bips/blob/4c6389f8431f677847b115538a47ce8c826c6be8/bip-0341.mediawiki#script-validation-rules.
+	switch net.Net {
+	case wire.MainNet, wire.TestNet3: // enable Taproot for Bitcoin mainnet/testnet.
+		if len(pkScript) == 34 && pkScript[0] == txscript.OP_1 && pkScript[1] == 0x20 {
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/coins/btc/util/util.go:72]: The text at the link says

>  Taproot output is a native SegWit output (see BIP141) with version number 1, and a 32-byte witness program.

Following the link (https://github.com/bitcoin/bips/blob/4c6389f8431f677847b115538a47ce8c826c6be8/bip-0141.mediawiki#witness-program), quote:

> A scriptPubKey (or redeemScript as defined in BIP16/P2SH) that consists of a 1-byte push opcode (for 0 to 16) followed by a data push between 2 and 40 bytes gets a new special meaning. The value of the first push is called the "version byte". The following byte vector pushed is called the "witness program".

OP_1 is the version byte for v1 (taproot) witness outputs.

The witness program is a 32 byte pubkey, but it is a data push. A data push is prefixed with the length of the data being pushed, in this case 0x20 (=32) bytes.

File: backend/coins/btc/coin.go [hunk 5]
```diff
@@ -31,6 +31,7 @@ import (
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/electrum"
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/headers"
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/coin"
+	coinpkg "github.com/digitalbitbox/bitbox-wallet-app/backend/coins/coin"
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/coins/btc/coin.go:34]: good find but completely unrelated to this PR? --> https://github.com/digitalbitbox/bitbox-wallet-app/pull/1555

File: backend/coins/btc/util/util_test.go [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPkScriptFromAddress(t *testing.T) {
+	hash := []byte("\x92\x95\x3b\x69\x91\x29\x70\x02\xfa\xa6\x2a\x1d\xd2\x43\x13\xff\x62\x1e\x10\xab")
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/coins/btc/util/util_test.go:27]: done. i wish this was in stdlib :(

File: backend/coins/btc/addresses/address.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (benma) [reply backend/coins/btc/addresses/address.go]: imported as `ourbtcutil` :shrug:

**Review by x1ddos — APPROVED** [review]
LGTM

**Pull request merged**